tests/test_device_validator.py

```diff
@@ -77,6 +77,18 @@ def test_load_device_setting(self):
             validate_load_device(fit_to_limit=True,
                                  energy_rate_increase_per_update=2)
 
+        with self.assertRaises(D3ADeviceException):
+            validate_load_device(avg_power_W=100,
+                                 daily_load_profile=[1, 2, 3, 4, 5, 6])
+
+        with self.assertRaises(D3ADeviceException):
+            validate_load_device(hrs_per_day=1,
+                                 daily_load_profile=[1, 2, 3, 4, 5, 6])
+
+        with self.assertRaises(D3ADeviceException):
+            validate_load_device(hrs_of_day=[1, 2, 3, 4, 5, 6],
+                                 daily_load_profile=[1, 2, 3, 4, 5, 6])
```

Comment:
One more check that would be nice to have a type check via ast.literal_eval is of `daily_load_profile`. If I am right `daily_load_profile`  should be in dict format having key as an operating hour and value as energy_demanded for that hour.

Reply:
Ok, will take the opportunity to add it.